cpg-core/src/main/java/de/fraunhofer/aisec/cpg/graph/TypeManager.java

```diff
@@ -336,13 +336,16 @@ public static void setTypeSystemActive(boolean active) {
   }
 
   @NotNull
-  public Map<HasType, Set<Type>> getTypeCache() {
+  public Map<HasType, List<Type>> getTypeCache() {
     return typeCache;
   }
 
   public synchronized void cacheType(HasType node, Type type) {
     if (!isUnknown(type)) {
-      typeCache.computeIfAbsent(node, n -> new HashSet<>()).add(type);
+      List<Type> types = typeCache.computeIfAbsent(node, n -> new ArrayList<>());
+      if (!types.contains(type)) {
```

Comment:
Isn't this essentially the same as a set does? Not sure this is really faster / better here.